spt/strafestuff.hpp

```diff
@@ -322,7 +322,23 @@ bool Strafe(PlayerData& player, const MovementVars& vars, bool onground, bool ju
 {
 	//DevMsg("[Strafing] ducking = %d\n", (int)ducking);
 	if (jumped && player.Velocity.Length2D() >= vars.Maxspeed * ((ducking || (vars.Maxspeed == 320)) ? 0.1 : 0.5)) {
-		out.Yaw = NormalizeDeg(tas_strafe_yaw.GetFloat() + 180);
+		if y_spt_glitchless_bhop_enabled = 1 {
+			const Vector vel = serverDLL.GetLastVelocity();
+			if vel.x >= 0 and vel.y >= 0 {
+				out.Yaw = NormalizeDeg(double atan(double vel.y/vel.x));
+			}
+			if vel.x <= 0 and vel.y >= 0 {
+				out.Yaw = NormalizeDeg(double atan(double vel.y/-vel.x)+90);
+			}
+			if vel.x <= 0 and vel.y <= 0 {
+				out.Yaw = NormalizeDeg(double atan(double -vel.y/-vel.x)+180);
+			}
+			if vel.x >= 0 and vel.y <= 0 {
+				out.Yaw = NormalizeDeg(double atan(double -vel.y/vel.x)-90);
+			}
+		}
+		else if y_spt_glitchless_bhop_enabled = 0 {
```

Comment:
Also `=` is an assignment. You want `==` for checks. Also, you need brackets around `if` conditions or the code won't compile.